### PR TITLE
Implemented multistage Dockerfile to shrink SOASuite image size

### PR DESCRIPTION
**Pull request by ericsmalling**
Final image size shrunk from 13.5GB to 6.37GB due to removal of intermediate installer file layers

### CODE REVIEW

**Comment by ericsmalling** [comment]
Similar to https://github.com/oracle/docker-images/pull/952 and https://github.com/oracle/docker-images/pull/946

**Comment by Djelibeybi** [comment]
@ericsmalling could you please sign (or point me towards your existing) [Oracle Contributor License Agreement](http://www.oracle.com/technetwork/community/oca-486395.html)?

**Comment by Djelibeybi** [comment]
I'm trying to find a new person from the SOA team to review this.

**Comment by mriccell** [comment]
Avi the new SOA person incharged of images is Raj Venkatesan

**Comment by Djelibeybi** [comment]
@ericsmalling still waiting on your Oracle CLA too.

File: OracleSOASuite/dockerfiles/12.2.1.3/Dockerfile [hunk 1]
```diff
@@ -31,20 +30,34 @@ ENV FMW_JAR1=fmw_12.2.1.3.0_soa.jar \
 # -----------------------------------
 ADD  $FMW_JAR1 $FMW_JAR2 /u01/
 COPY container-scripts/* /u01/oracle/dockertools/
+COPY install/* /u01/
+
+USER root
 RUN  yum install -y hostname &&    \
```
Comment (rdas76) [inline OracleSOASuite/dockerfiles/12.2.1.3/Dockerfile]: The following 2 lines can be removed from here as these are duplicate and we need to run the yum install later.

File: OracleSOASuite/dockerfiles/12.2.1.3/Dockerfile [hunk 2]
```diff
@@ -31,20 +30,34 @@ ENV FMW_JAR1=fmw_12.2.1.3.0_soa.jar \
...
      rm -rf /var/cache/yum &&      \
      cd /u01 && chmod 755 *.jar && \
+     chown oracle:oracle /u01/*.jar /u01/*.response /u01/*.loc && \
      chmod a+xr /u01/oracle/dockertools/*.*
```
Comment (rdas76) [inline OracleSOASuite/dockerfiles/12.2.1.3/Dockerfile]: This line can also be removed as it is duplicate here.

File: OracleSOASuite/dockerfiles/12.2.1.3/Dockerfile [hunk 3]
```diff
@@ -31,20 +30,34 @@ ENV FMW_JAR1=fmw_12.2.1.3.0_soa.jar \
...
-  $JAVA_HOME/bin/java -jar $FMW_JAR2 -silent -responseFile /u01/soasuite.response -invPtrLoc /u01/oraInst.loc -jreLoc $JAVA_HOME -ignoreSysPrereqs -force -novalidation ORACLE_HOME=$ORACLE_HOME INSTALL_TYPE="Service Bus" && \
-  rm -fr /u01/*.jar /u01/*.response /u01/*.loc
+WORKDIR /u01
+RUN $JAVA_HOME/bin/java -jar $FMW_JAR1 -silent -responseFile /u01/soasuite.response -invPtrLoc /u01/oraInst.loc -jreLoc $JAVA_HOME -ignoreSysPrereqs -force -novalidation ORACLE_HOME=$ORACLE_HOME
+RUN $JAVA_HOME/bin/java -jar $FMW_JAR2 -silent -responseFile /u01/soasuite.response -invPtrLoc /u01/oraInst.loc -jreLoc $JAVA_HOME -ignoreSysPrereqs -force -novalidation ORACLE_HOME=$ORACLE_HOME INSTALL_TYPE="Service Bus"
+
+USER root
+RUN rm -fr /u01/*.jar /u01/*.response /u01/*.loc
 
 #
```
Comment (rdas76) [inline OracleSOASuite/dockerfiles/12.2.1.3/Dockerfile]: This comment to go before CMD at the bottom of the file.

File: OracleSOASuite/dockerfiles/12.2.1.3/Dockerfile (hunk 1 above)
Comment (rdas76) [reply OracleSOASuite/dockerfiles/12.2.1.3/Dockerfile]: Referring to 

RUN  yum install -y hostname &&    \
     rm -rf /var/cache/yum &&      \

File: OracleSOASuite/dockerfiles/12.2.1.3/Dockerfile (hunk 2 above)
Comment (rdas76) [reply OracleSOASuite/dockerfiles/12.2.1.3/Dockerfile]: Referring to 

chmod a+xr /u01/oracle/dockertools/*.*

**Comment by Djelibeybi** [comment]
Closing due to lack of response.

**Pull request closed without merging**